vsintegration/src/FSharp.LanguageService/ProjectSitesAndFiles.fs

```diff
@@ -231,93 +235,120 @@ type internal ProjectSitesAndFiles() =
+        let getReferencesForSolutionService (solutionService:IVsSolution) =
+            [|
+                match referencedProjects projectSite, extraProjectInfo with
+                | None, Some (:? VisualStudioWorkspaceImpl as workspace) when not (isNull workspace.CurrentSolution)->
                     let path = projectSite.ProjectFileName()
-                    let projectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(path, projectDisplayNameOf path)
-                    let project = workspace.CurrentSolution.GetProject(projectId)
-                    if not (isNull project) then
-                        for reference in project.ProjectReferences do
-                            let project = workspace.CurrentSolution.GetProject(reference.ProjectId)
-                            let siteProvider = provideProjectSiteProvider (workspace, project, serviceProvider, projectOptions)
-                            yield (Path.GetFileName(project.FilePath), Path.GetFullPath project.OutputFilePath, siteProvider)
-                | _ -> ()
-        }
-
-    static let rec referencedProjectsOf(enableInMemoryCrossProjectReferences, tryGetOptionsForReferencedProject, projectSite, serviceProvider, extraProjectInfo, projectOptions, useUniqueStamp) =
-        [| for (projectFileName, projectPath, projectSiteProvider) in referencedProvideProjectSites (projectSite, serviceProvider, extraProjectInfo, projectOptions) do
+                    if not (String.IsNullOrWhiteSpace(path)) then
+                        let projectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(path, projectDisplayNameOf path)
+                        let project = workspace.CurrentSolution.GetProject(projectId)
+                        if not (isNull project) then
+                            for reference in project.ProjectReferences do
+                                let project = workspace.CurrentSolution.GetProject(reference.ProjectId)
+                                let siteProvider = provideProjectSiteProvider (workspace, project, serviceProvider, projectOptionsTable)
+                                let outputPath =
+                                    if not (String.IsNullOrWhiteSpace(project.OutputFilePath)) then
+                                        project.OutputFilePath
+                                    else
+                                        // ======================================================================
+                                        // Giant Hack here !!!!!!
+                                        // project.OutputFilePath does not have a value at this point ... 
+                                        // so calculate one by peeking at the command line options ... --out:
+                                        // ======================================================================
+                                        match projectOptionsTable with
```

Comment:
I don't understand why this works.  According to what @davkean was telling me yesterday, at the point where the OutputFilePath is still null, we shouldn't have any options yet, just the source files because it comes from evaluation only...

Reply:
@kevinpi.  Well the command line options from designtimebuild have been received at this point.